src/main/java/baritone/pathing/movement/movements/MovementParkourAdv.java

```diff
@@ -0,0 +1,1054 @@
+                }
+                return WALK_ONE_BLOCK_COST * distance + context.jumpPenalty;
+            case MOMENTUM:
+                return SPRINT_ONE_BLOCK_COST * distance * 2 + context.jumpPenalty * 2; // Momentum jumps are unsafe, therefore should have high costs.
+            case EDGE_NEO:
+                return SPRINT_ONE_BLOCK_COST * distance + context.jumpPenalty * 2;
+        }
+        return 0; //Will never reach this unless a new type is added
+    }
+
+    private final EnumFacing destDir = EnumFacing.fromAngle(MathHelper.atan2(src.x - dest.x, -(src.z - dest.z)) * RotationUtils.RAD_TO_DEG);
+
+
+    @Override
+    protected boolean prepared(MovementState state) {
+        if (inStartingPosition || state.getStatus() == MovementStatus.WAITING) {
+            return true;
+        }
+        Vec3d offset = new Vec3d(simpleDirection.getOpposite().getDirectionVec()).scale(PREP_OFFSET); // defaults for a NORMAL JumpType
+        double accuracy = 0.1; // defaults for a NORMAL JumpType
+        switch (type) {
+            case MOMENTUM:
+                accuracy = 0.025; // Basically as small as you can get
+                break;
+            case EDGE:
+                offset = new Vec3d(simpleDirection.getDirectionVec()).scale(0.8);
+                accuracy = 0.025; //Don't want to fall off
+                break;
+            case EDGE_NEO:
+                offset = new Vec3d(simpleDirection.getDirectionVec()).scale(0.8).add(new Vec3d(destDir.getOpposite().getDirectionVec()).scale(0.2));
+                accuracy = 0.025;
+                break;
+        }
+        Vec3d preJumpPos = offset.add(src.x + 0.5, ctx.player().posY, src.z + 0.5);
+        double distance = preJumpPos.distanceTo(ctx.playerFeetAsVec());
+        //System.out.println("Distance to prepLoc = " + distance);
+        boolean prepLocPassable = MovementHelper.fullyPassable(ctx, src.add(new BlockPos(offset.add(offset.normalize().scale(0.4))))); // Checking 0.4 blocks in the direction of offset for a block (0.3 is the player hitbox width)
+        if (((distance > accuracy && prepLocPassable) || (distance > (PREP_OFFSET - (0.2 - accuracy)) && !prepLocPassable)) &&
+                (atDestTicks < 7 || accuracy >= 0.1)) { // Accuracies over 0.1 will require additional wait ticks to reduce excess momentum
+            if (atDestTicks < 5) {
+                MovementHelper.moveTowards(ctx, state, offset.add(VecUtils.getBlockPosCenter(src)));
+            }
+            if(distance < 0.2) {
+                state.setInput(Input.SNEAK, true);
+                if(distance < accuracy) {
+                    atDestTicks++;
+                } else {
+                    atDestTicks--;
+                }
+            } else {
+                atDestTicks = 0;
+            }
+            if (ctx.playerFeet().y < src.y) {
+                // we have fallen
+                logDebug("sorry");
```

Comment:
have something more helpful in the log

Reply:
Fixed. (I was using code from MovementParkour as my base)